sys/include/net/gnrc/gnrc_mac/packet_queue.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @ingroup     net_gnrc_mac
+ * @file
+ * @brief       Wrapper for priority_queue that holds gnrc_pktsnip_t* and is
+ *              aware of it's length.
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ * @}
+ */
+
+#ifndef GNRC_MAC_PACKET_QUEUE_H
+#define GNRC_MAC_PACKET_QUEUE_H
+
+#include <stdint.h>
+#include <priority_queue.h>
+#include <net/gnrc/pkt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef priority_queue_node_t packet_queue_node_t;
+
+/* TODO: Description */
+typedef struct {
+    priority_queue_t queue;
+    uint32_t length;
+    packet_queue_node_t* buffer;
+    size_t buffer_size;
+} packet_queue_t;
+
```

Comment:
I think this packet_queue module here is still not ready for universal usage yet. Just my opinion. This is because, if you check the packet_queue_t structure above, it contains a buffer (type of packet_queue_node_t, which is actually priority_queue_node_t). This means that you need to define a packet buffer first for holding your packets before using queue management. 
In short, the queue management module here relies on a predefined packet buffer. check the example case in Lw-MAC: [buffer define](https://github.com/RIOT-OS/RIOT/pull/3730/files#diff-21fdd181b028f70dc7c1069e53d8d8bbR112), [init function](https://github.com/RIOT-OS/RIOT/pull/3730/files#diff-4afd53687427cf8e699e1f899a2fedbbR57) and [init before use](https://github.com/RIOT-OS/RIOT/pull/3730/files#diff-2a301ee7ac4929f1cfad9baee9e23ce2R109).

So, currently it is more MAC-oriented... and I tend to revise it after the merge of the Lw-MAC.